src/pages/Country.jsx

```diff
@@ -0,0 +1,122 @@
+import { useTheme } from '../hooks/themeUtils';
+import { useParams, useNavigate, Link } from 'react-router-dom';
+import { useState, useEffect } from 'react';
+import { API_URL2 } from '../config.js';
+import style from './Country.module.scss';
+import Preloader from '../components/preloader/Preloader';
+import IconArrowLeft from '../assets/images/arrow-left-svgrepo-com.svg?react';
+import { BorderCountry } from '../components/BorderCountry';
+
+function Country() {
+  const { id } = useParams();
+  const navigate = useNavigate();
+  const [country, setCountry] = useState(null);
+  const [isLoading, setLoading] = useState(false);
+  const { theme = 'dark' } = useTheme();
+  const themeIconArrow = <IconArrowLeft stroke={theme === 'light' ? 'white' : 'black'} />;
+  const CountryStyle = `${style.card} ${theme === 'dark' ? ` ${style.light}` : `''`}`;
+  const CountryStyleBtn = `${style.btn} ${theme === 'dark' ? ` ${style.light}` : `''`}`;
+  const CountryStyleCardInfo = `${style.cardInfo} ${theme === 'dark' ? ` ${style.light}` : `''`}`;
+  const CountryStyleBtnBlock = `${style.buttonBlock} ${theme === 'dark' ? ` ${style.light}` : `''`}`;
+
+  const goBack = () => navigate(-1);
+  const currencies = country
+    ? Object.values(country.currencies)
+        .filter((currency) => Boolean(currency.name))
+        .map((cur) => cur.name)
+        .join(', ')
+    : '';
+  const languages = country ? Object.values(country.languages).join(', ') : '';
+
+  const nativeName = country
+    ? Object.values(country.name.nativeName)
+        .filter((name) => Boolean(name.official))
+        .map((natName) => natName.official)
+        .join(', ')
+    : '';
+
+  useEffect(() => {
+    fetch(API_URL2 + id)
+      .then((response) => response.json())
+      .then((data) => {
+        setCountry(data[0]);
+        setLoading(true);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }, [id]);
+
+  console.log(country);
+
+  return (
+    <div className={CountryStyle}>
+      <button className={CountryStyleBtn} onClick={goBack}>
+        {themeIconArrow}Back
+      </button>
+      {!isLoading ? (
+        <Preloader />
+      ) : country ? (
```

Comment:
Semantically it looks wrong. 

You're saying: "When it isn't loading then show preloader. But when it's actually loading show me the content.

Reply:
Man, I see you've updated the code, but the sematically it's still broken.

You should show Preloader when isLoading is true. You did that before.

Reply:
oops